.github/workflows/pypi-build.yml

```diff
@@ -0,0 +1,44 @@
+on:
+  pull_request:
+    branches: [ "main" ]
+jobs:
+  pypi-publish:
+    name: Upload release to PyPI
+    runs-on: ubuntu-latest
+    environment:
+      name: pypi
+      url: https://pypi.org/project/causica
+    permissions:
+      id-token: write  # IMPORTANT: this permission is mandatory for trusted publishing
+
+    steps:
+    - uses: actions/checkout@v3
+
+    - uses: actions/setup-python@v2
+      with:
+        python-version: "3.10"
+
+    - uses: actions/cache@v3
+      with:
+        path: ~/.cache/pypoetry/virtualenvs
+        key: ${{ hashFiles('poetry.lock') }}
+      id: cache
+
+    - name: Install poetry
+      run: |
+        curl -sSL https://install.python-poetry.org | python3 -
+
+    - name: Install dependencies
+      run: |
+        poetry env use 3.10
+        poetry build
+
+    - name: Upload build artifact
+      uses: actions/upload-artifact@v3
+      with:
+        name: Build artifacts
+        path: ./dist
+        retention-days: 90
+
+    - name: Publish package distributions to PyPI
+      uses: pypa/gh-action-pypi-publish@release/v1
```

Comment:
\+`\n`

Reply:
if it makes you happy

Reply:
GitHub and I rejoice: 
![image](https://github.com/microsoft/causica/assets/2157859/4320005f-48cb-456f-b0d0-9046a7c59b18)